tools/licenses/lib/licenses.dart

```diff
@@ -752,13 +756,16 @@ Iterable<_LicenseMatch> _tryReferenceByFilename(String body, LicenseFileReferenc
   }
 }
 
-Iterable<_LicenseMatch> _tryReferenceByType(String body, RegExp pattern, LicenseSource parentDirectory, { String origin }) sync* {
-  for (_PartialLicenseMatch match in _findLicenseBlocks(body, pattern, 1, 2)) {
+Iterable<_LicenseMatch> _tryReferenceByType(String body, RegExp pattern, LicenseSource parentDirectory, { String origin, bool needsCopyright: true }) sync* {
+  for (_PartialLicenseMatch match in _findLicenseBlocks(body, pattern, 1, 2, needsCopyright: needsCopyright)) {
     final LicenseType type = convertLicenseNameToType(match.group(3));
     final License template = parentDirectory.nearestLicenseOfType(type);
     if (template == null)
       throw 'failed to find accompanying $type license in $parentDirectory';
-    assert(_reformat(match.getCopyrights()) != '');
+    if (needsCopyright)
+      assert(_reformat(match.getCopyrights()) != '');
```

Comment:
unless you really mean to just assert, in which case the whole four lines should all be in an assert block.